### ISSUE TITLE
Dino cannot encrypt messages sent while offline

### ISSUE THREAD

**Issue by devurandom** [dino/dino#440]
Dino cannot decrypt messages that were sent while it was offline. This affects all of:
* Someone else sends a message to my account
* I send a message from Conversations to someone else (and Dino receives it via CC)
* I send a message from Dino on another computer to someone else (and this Dino receives it via CC)

It affects both Dino installations on my two computers.

Dino displays these messages as:
* either `I sent you an OMEMO encrypted message but your client doesn’t seem to support that.`
* or `[This message is OMEMO encrypted]`

Dino version, desktop: ee5c838a6ba85c9d6e72ac2941a7994505aaa4fc
Dino version, laptop: ee5c838a6ba85c9d6e72ac2941a7994505aaa4fc
Conversations version, mobile phone: (latest from Google Play, will add information later)
Prosody server version: 0.10.2
  - Prosody modules version: 2853:a844d1535c4d

**Comment by licaon-kter** [comment]
@devurandom Server setup 100% in compliance?

**Comment by devurandom** [comment]
> @devurandom Server setup 100% in compliance?

Compliance to what? And how would I test that?

**Comment by licaon-kter** [comment]
https://github.com/iNPUTmice/caas or maybe it's already listed in http://compliance.conversations.im/ ?

**Comment by devurandom** [comment]
> https://github.com/iNPUTmice/caas or maybe it's already listed in http://compliance.conversations.im/ ?

I had some issues getting it to run on my system (iNPUTmice/caas#23), but now I have the results:
```
Server is running Prosody 0.10.3

Compliance report for kynes.de
XEP-0313: Message Archive Management (for Multi-User Chat)       PASSED
XEP-0191: Blocking Command                                       PASSED
XEP-0363: HTTP File Upload                                       PASSED
Roster Versioning                                                PASSED
XEP-0115: Entity Capabilities                                    PASSED
XEP-0313: Message Archive Management                             PASSED
XEP-0160: Best Practices for Handling Offline Messages           PASSED
XEP-0163: Personal Eventing Protocol                             PASSED
XEP-0368: SRV records for XMPP over TLS                          FAILED
XEP-0065: SOCKS5 Bytestreams (Proxy)                             PASSED
XEP-0280: Message Carbons                                        PASSED
XEP-0045: Multi-User Chat                                        PASSED
XEP-0153: vCard-Based Avatar (MUC)                               FAILED
XEP-0384: OMEMO Encryption                                       PASSED
XEP-0352: Client State Indication                                PASSED
XEP-0357: Push Notifications                                     PASSED
XEP-0398: User Avatar to vCard-Based Avatars Conversion          PASSED
XEP-0198: Stream Management                                      PASSED

Informational tests:
Contact Addresses for reporting abuse                            FAILED
XEP-0156: Discovering Alternative XMPP Connection Methods (HTTP) PASSED
XEP-0077: In-Band Registration                                   FAILED
XEP-0363: HTTP File Upload (CORS Headers)                        FAILED
```

I.e. it is 100% in compliance regarding OMEMO.

**Comment by licaon-kter** [comment]
Ok, now... The OP issue?

**Comment by devurandom** [comment]
Which issue?

**Comment by licaon-kter** [comment]
@devurandom "OP" means "opening post"... You've setup he server...ok...the issue was about offline messages and such. So?

**Comment by devurandom** [comment]
@licaon-kter Yes, please continue helping me to debug the issue mentioned in the opening post, if you now have all the information you require.

**Comment by main--** [comment]
Not sure if this changed since the issue was opened but for me, dino simply doesn't show messages I sent from other devices. Server is compliant, key is trusted. Not sure what my mistake is. It works with other clients.

**Comment by main--** [comment]
To clarify, plaintext messages work just fine. It's only my own OMEMO messages (the ones I send from my other devices) that won't show up in dino.

**Comment by devurandom** [comment]
> To clarify, plaintext messages work just fine. It's only my own OMEMO messages (the ones I send from my other devices) that won't show up in dino.

By now I think we actually have multiple issues here. Currently, messages by other people *to* me do not show up *at all* in Dino, while messages send *by* me do. Messages sent while I was offline also show up -- my own and those from my chat partners. I guess the first step would be error messages, so we can figure out whether Dino actually sees the messages and then discards them.

**Comment by moppman** [comment]
Maybe related to https://github.com/dino/dino/issues/473 ?

**Comment by selurvedu** [comment]
> Currently, instead of the decryption problem, messages by other people _to me_ do not show up _at all_ in Dino, while messages send _by me_ do. But when _I write_ a message from another device and my chat _partner replies_ to that, his message also **also** does not show up at all in Dino. Afterwards, messages sent from Dino _by me_ **cannot be decrypted** _by my chat partner_ anymore.
> 
> I guess the first step would be error messages, so we can figure out whether Dino actually sees the messages and then discards them.

Same problem here. It's been around for 2 years or so.

**Comment by licaon-kter** [comment]
@selurvedu What other clients are involved? Are your keys trusted there?

**Comment by selurvedu** [comment]
@licaon-kter, recently I tested sending messages between Dino, Gajim, Pidgin 2 + lurch, Conversations and aTalk.

The only problem I had was Dino and lurch being unable to communicate with each other properly.

Also, when I tried purging unused identities in Gajim, it stopped decrypting messages sent from Dino, however, I couldn't reproduce that bug.
